backend/tests/test_graphql_api/test_recurring_reservation/test_add_reservation.py

```diff
@@ -78,76 +80,154 @@ def test_recurring_reservations__add_reservation__begin_after_end(graphql):
-    recurring_reservation = create_reservation_series(
+    series = create_reservation_series(
         reservation_unit__access_types__access_type=AccessType.ACCESS_CODE,
         reservations__access_type=AccessType.ACCESS_CODE,
         reservations__access_code_is_active=True,
         reservations__access_code_generated_at=local_datetime(2024, 1, 1),
     )
 
-    assert recurring_reservation.reservations.count() == 9
-    existing = list(recurring_reservation.reservations.values_list("pk", flat=True))
+    assert series.reservations.count() == 9
+    existing = list(series.reservations.values_list("pk", flat=True))
 
-    data = get_minimal_add_data(recurring_reservation)
+    data = get_minimal_add_data(series)
 
     graphql.login_with_superuser()
     response = graphql(ADD_RESERVATION_TO_SERIES_MUTATION, input_data=data)
 
     assert response.has_errors is False
 
-    assert recurring_reservation.reservations.count() == 10
-    new_reservation: Reservation | None = recurring_reservation.reservations.exclude(pk__in=existing).first()
+    assert series.reservations.count() == 10
+    new_reservation: Reservation | None = series.reservations.exclude(pk__in=existing).first()
 
     assert new_reservation is not None
 
     assert new_reservation.access_type == AccessType.ACCESS_CODE
 
-    # These are the default, will be updated by rescheduling.
+    # These are the default, would be updated by rescheduling, but we mocked it out.
     assert new_reservation.access_code_is_active is False
     assert new_reservation.access_code_generated_at is None
 
     assert PindoraService.reschedule_access_code.called is True
 
 
+@freeze_time(local_datetime(2024, 1, 1))
+@patch_method(PindoraService.reschedule_access_code, side_effect=PindoraNotFoundError("Not Found"))
+@patch_method(PindoraService.create_access_code)
+def test_recurring_reservations__add_reservation__access_code__create_if_not_found(graphql):
+    series = create_reservation_series(
+        reservation_unit__access_types__access_type=AccessType.ACCESS_CODE,
+        reservations__access_type=AccessType.ACCESS_CODE,
+        reservations__access_code_is_active=True,
+        reservations__access_code_generated_at=local_datetime(2024, 1, 1),
+    )
+
+    assert series.reservations.count() == 9
+    existing = list(series.reservations.values_list("pk", flat=True))
+
+    data = get_minimal_add_data(series)
+
+    graphql.login_with_superuser()
+    response = graphql(ADD_RESERVATION_TO_SERIES_MUTATION, input_data=data)
+
+    assert response.has_errors is False
+
+    assert series.reservations.count() == 10
+    new_reservation: Reservation | None = series.reservations.exclude(pk__in=existing).first()
+
+    assert new_reservation is not None
+
+    assert new_reservation.access_type == AccessType.ACCESS_CODE
+
+    # These are the default, will be updated in the background task, since the rescheduling failed.
```

Comment:
... will be updated by create, since rescheduling did not find existing access code